### PR TITLE
quick-der library, asn2quickder + hexio tools

### PR DESCRIPTION
**Pull request by leenaars**
###### Motivation for this change

Libraries and tools used in ARPA2 project.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [x] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/asn1ate/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchFromGitHub, python, pythonPackages, buildPythonPackage, self }:
```
Comment (FRidh) [inline pkgs/development/tools/asn1ate/default.nix]: pythonPackages has python and buildPythonPackage as attributes. Could you remove those two here and use either a `let` expression or `with` instead.

File: pkgs/development/tools/asn1ate/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+let
+  pname = "asn1ate";
+  owner = "kimgr";
+in
+
+buildPythonPackage rec {
+  name = "${pname}-${version}";
+  version = "c56104e8912400135509b584d84423ee05a5af6b";
```
Comment (FRidh) [inline pkgs/development/tools/asn1ate/default.nix]: Please check the latest version of the coding conventions in the Nixpkgs manual. When not using a release the version needs to be a date, and the suffix `-unstable` is added to the name. So e.g. `asn1ate-unstable-2016-08-10`.

File: pkgs/development/tools/asn1ate/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    sha256 = "04pddr1mh2v9qq8fg60czwvjny5qwh4nyxszr3qc4bipiiv2xk9w";
+    rev = "${version}";
+    owner = "kimgr";
+    repo = pname;
+  };
+
+  propagatedBuildInputs = [ python pythonPackages.pyparsing ];
+
+  meta = with stdenv.lib; {
+    description = "Python library for translating ASN.1 into other forms";
```
Comment (FRidh) [inline pkgs/development/tools/asn1ate/default.nix]: if this is a library, then it should be added to `python-packages.nix` instead.

File: pkgs/development/tools/asn2quickder/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchFromGitHub, python, pythonPackages, buildPythonPackage, makeWrapper }:
```
Comment (FRidh) [inline pkgs/development/tools/asn2quickder/default.nix]: as mentioned before

File: pkgs/development/tools/asn1ate/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  version = "c56104e8912400135509b584d84423ee05a5af6b";
+
+  src = fetchFromGitHub {
+    sha256 = "04pddr1mh2v9qq8fg60czwvjny5qwh4nyxszr3qc4bipiiv2xk9w";
+    rev = "${version}";
+    owner = "kimgr";
+    repo = pname;
+  };
+
+  propagatedBuildInputs = [ python pythonPackages.pyparsing ];
```
Comment (FRidh) [inline pkgs/development/tools/asn1ate/default.nix]: no need to add python here because it is added by `buildPythonPackage`

**Comment by Mic92** [comment]
I changed/updated your pull request slightly here: https://github.com/Mic92/nixpkgs/tree/pr17579
Does this still work for you?

**Comment by leenaars** [comment]
Yes, it does. Thanks for the rebase!

**Comment by Mic92** [comment]
I rebased [again](9d6477bb7bd614b310a377e78a6af666f5b4d387), however I noticed that quickder does install any executable, but you created bin, sbin ... is this intented?

**Comment by leenaars** [comment]
You are completely right. Those can actually be removed. Thanks for noticing.

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/quickder/default.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  version = "0.1-RC12";
+
+  src = fetchFromGitHub {
+    sha256 = "0dh4gqhfbw041gxkwn48l2s438awng302ndff364dl4adwllk2nc";
+    rev = "version-${version}";
+    owner = "vanrein";
+    repo = "quick-der";
+  };
+
+  phases = [ "unpackPhase" "patchPhase" "installPhase" ];
```
Comment (leenaars) [inline pkgs/development/libraries/quickder/default.nix]: Have removed them, and updated the packages to their latest version.

Comment (leenaars) [inline pkgs/development/libraries/quickder/default.nix]: Also rebased, of course.

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**